File: src/Baekjoon_3052.java
import java.util.Scanner;
import java.util.Arrays;
public class Baekjoon_3052 {
    public static void main(String[] args) {
        Scanner input=new Scanner(System.in);
        int[] array=new int[10];
        int x=0;
        for(int i=0;i<10;i++){
            int num=input.nextInt();
            array[i]=num%42;
        }
        Arrays.sort(array);
        int result=1;
        for(int i=0;i<9;i++){
            if(array[i]!=array[i+1]){
                result++;
            }
        }

        System.out.println(result);
    }
}

File: 1차원 배열/BAEKJOON_3052.py
num, t, total=0, 0, 0

k=[]

for i in range(1, 11):
    num=int(input())

    t=num%42
    k.append(t)


for f in range(0, 42):
    s=k.count(f)
    if s>=1:
        total+=1

print(total)